### PR TITLE
fix(ENTESB-11578): Improve conditional flow usability

### PR DESCRIPTION
**Pull request by christophd**
Adds a select config mode (basic, advanced) page to step configuration so user can choose how to provide condition expressions.

Fixes ENTESB-11578

### CODE REVIEW

File: app/ui-react/syndesis/src/modules/integrations/components/editor/EditorApp.tsx [hunk 1]
```diff
@@ -207,15 +206,14 @@ export const EditorApp: React.FunctionComponent<IEditorApp> = ({
     <DescribeChoiceDataShapePage
       cancelHref={cancelHref}
       sidebar={props => (
-        <EditorSidebar {...props} isAdding={false} />
+        <EditorSidebar {...props} isAdding={mode === 'adding'} />
```
Comment (riccardo-forina) [inline app/ui-react/syndesis/src/modules/integrations/components/editor/EditorApp.tsx]: Good catch

File: app/ui-react/syndesis/src/modules/integrations/components/editor/choice/SelectChoiceConfigModePage.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+ * [state]{@link ISelectConfigModeRouteState} to be properly set in the route
+ * object.
+ *
+ * **Warning:** this component will throw an exception if the route state is
+ * undefined.
+ */
+export class SelectChoiceConfigModePage extends React.Component<ISelectChoiceConfigModePageProps> {
+  public render() {
+    return (
+      <WithRouteData<ISelectConfigModeRouteParams, ISelectConfigModeRouteState>>
```
Comment (riccardo-forina) [inline app/ui-react/syndesis/src/modules/integrations/components/editor/choice/SelectChoiceConfigModePage.tsx]: For the future, there is an equivalent hook for this called `useRouteData`. No need to change this now, eventually we'll update the whole routing thing when `react-router` will hit v6.

**Comment by dongniwang** [comment]
Hey @christophd - thanks for the PR. Would it be possible to include some key screenshots? Staging is down at the moment for me...and I'd like to take a look at it as soon as possible. Thank you!

**Comment by christophd** [comment]
@dongniwang I have added a screencast to the ENTESB issue

**Pull request merged**